Upgrade Swagger UI

pluma opened the pull request:
This PR upgrades Swagger UI to the latest version.

pluma commented on js/server/assets/swagger/index.html:
```diff
@@ -1,132 +1,92 @@
+        layout: "StandaloneLayout",
+        docExpansion: "list",
+        requestInterceptor: function (request) {
+          if (!request.headers.Authorization && !request.headers.authorization) {
```
FYI this is a change in our code: instead of always setting this header we now only set it if not already present. This avoids issues later on caused by accidentally overriding the user-specified value.

hkernbach replied:
> This avoids issues later on caused by accidentally overriding the user-specified value.
Somehow we need to be sure, that we do not throw away that change in the future

hkernbach replied:
@pluma could we maybe write a dumb test which will fail if we update without keeping that change?

pluma replied:
I don't see how. This code is only executed if a user opens the swagger-ui in the browser and performs an operation. That'd require browser instrumentation to test, which we don't have and don't want.

The upstream swagger-ui code doesn't include the `requestInterceptor` config at all so updating without touching this would visibly break by making swagger-ui no longer load any specs at all and show a big scary error message.

The change my comment refers to is wrapping our code in an `if` to avoid clobbering user-provided headers (via the swagger-ui auth GUI).

As long as everyone who upgrades swagger-ui in our code follows the established practice, we'll be fine:

1. Find out which swagger-ui version we are currently using
2. Diff our index.html against the index.html bundled with that version by default
3. Replace our swagger-ui with the latest distribution
4. Manually re-apply the diff as appropriate

Yes, this is finicky, but this is the same process we've had for upgrading Swagger UI from the start.

dothebart replied:
please add this procedure to https://github.com/arangodb/arangodb/blob/devel/3rdParty/README_maintainers.md

pluma replied:
@dothebart sounds good to me. I'll add it in this PR.

pluma commented:
@dothebart I've added the steps necessary to update swagger-ui to the 3rdParty `README_maintainers.md` and added a comment to the top of the `index.html` indicating which upstream version the files were copied from.

@hkernbach please re-review given these changes.

dothebart commented on 3rdParty/README_maintainers.md:
```diff
@@ -123,6 +123,27 @@ https://github.com/google/snappy
+4. JS: some logic to inject the `jwt` as fallback auth via `requestInterceptor`
+5. An HTML comment at the start of the file indicating the swagger-ui version
+
+To determine which exact changes you need to make to `index.html`
```
please add the full path to this specific index.html here, and the upstream sample one.

pluma replied:
The files live directly in the only two folders mentioned in the first sentence of this section. I guess I can restructure the explanation to eliminate any ambiguity.

dothebart commented on 3rdParty/README_maintainers.md:
```diff
@@ -123,6 +123,27 @@ https://github.com/google/snappy
+To determine which exact changes you need to make to `index.html`
+compare the file you're replacing against the upstream version it was based on.
+
+Note that the changes we made need to be re-applied manually as details like
```
how to revalidate?

pluma replied:
The only way to test the swagger-ui is by opening it in a browser and clicking enough things until you're confident it's still working. Defining adequate acceptance criteria for "working" seems too much of a bikeshed for this PR to address right now. I'll add something to that effect for now.
